### PR TITLE
Upgrade agent service (Pydantic v2)

### PR DESCRIPTION
**Pull request by giancarloromeo**
## What do these changes do?
Upgrades the agent service to Pydantic v2.

## Related issue/s

- resolves #4481 

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/agent/tests/conftest.py [hunk 1]
```diff
@@ -187,9 +188,8 @@ def caplog_info_debug(
 
 
 @pytest.fixture(scope="module")
-def mocked_s3_server_url(mocked_aws_server: ThreadedMotoServer) -> HttpUrl:
+def mocked_s3_server_url(mocked_aws_server: ThreadedMotoServer) -> HttpUrlLegacy:
```
Comment (giancarloromeo) [inline services/agent/tests/conftest.py:191]: This `HttpUrlLegacy` has been introduced to be compliant with the previous `HttpUrl` (Pydantic v1). It solves the issue with the trailing slash.

See: https://docs.pydantic.dev/latest/migration/#url-and-dsn-types-in-pydanticnetworks-no-longer-inherit-from-str

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks! looks good.

File: services/agent/requirements/_base.txt [hunk 2]
```diff
@@ -41,16 +43,10 @@ click==8.1.3
     #   uvicorn
 dnspython==2.2.1
     # via email-validator
-email-validator==1.3.0
+email-validator==2.2.0
     # via pydantic
-fastapi==0.96.0
+fastapi==0.115.0
```
Comment (pcrespov) [inline services/agent/requirements/_base.txt:48]: OK so we also upgrade fastapi to the latest?

**Review by pcrespov — APPROVED** [review]
thx. go, go, go ... :-)

File: services/agent/requirements/_base.txt (hunk 2 above)
Comment (giancarloromeo) [reply services/agent/requirements/_base.txt:48]: Yup!

**Review by GitHK — APPROVED** [review]

**Pull request merged**